File: backend/src/main/java/de/interaapps/pastefy/model/requests/paste/CreatePasteRequest.java
package de.interaapps.pastefy.model.requests.paste;

import de.interaapps.pastefy.model.database.Paste;
import org.javawebstack.validator.rule.RequiredRule;

import java.util.List;

public class CreatePasteRequest {

    public String title = "";
    @RequiredRule
    public String content;
    public boolean encrypted = false;
    public String folder;

    public String expireAt;

    public String forkedFrom;

    public List<String> tags = null;

    public Paste.Visibility visibility = Paste.Visibility.UNLISTED;

    public Paste.Type type = Paste.Type.PASTE;

    public boolean ai = false;
}

File: backend-kt/src/main/kotlin/de/interaapps/pastefy/dto/pastes/CreatePasteRequest.kt
package de.interaapps.pastefy.dto.pastes

import de.interaapps.pastefy.enums.PasteType
import de.interaapps.pastefy.enums.PasteVisibility
import jakarta.validation.constraints.NotBlank

data class CreatePasteRequest(
    val title: String = "",
    @field:NotBlank
    val content: String,

    val encrypted: Boolean = false,
    val folder: String? = null,
    val expireAt: String? = null,
    val forkedFrom: String? = null,
    val tags: List<String>? = null,
    val visibility: PasteVisibility = PasteVisibility.UNLISTED,
    val type: PasteType = PasteType.PASTE,
    val ai: Boolean = false,
)
